File: minion/app/src/main/java/snakesoft/minion/Activities/WpActivity.java
package snakesoft.minion.Activities;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.TextView;

import snakesoft.minion.Models.GlobalModel;
import snakesoft.minion.R;

public class WpActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_wp);

        int successAtempts = GlobalModel.getWpModel().getSuccessAtempts();
        int failedAtempts = GlobalModel.getWpModel().getFailedAtempts();

        setData(successAtempts, failedAtempts);

        mBtnSuccess = (Button) findViewById(R.id.btnSuccess);
        mBtnSuccess.setOnClickListener(WpActivity.this);

        mBtnFailed = (Button) findViewById(R.id.btnFail);
        mBtnFailed.setOnClickListener(WpActivity.this);

        mBtnClear = (Button) findViewById(R.id.btnClear);
        mBtnClear.setOnClickListener(WpActivity.this);
    }

    private void setData(int successAtempts, int failedAtempts) {
        TextView tv = (TextView) findViewById(R.id.tvScore);
        tv.setText(Integer.toString(successAtempts) + ":" + Integer.toString(failedAtempts));

        ProgressBar pb = (ProgressBar) findViewById(R.id.progressBar);
        pb.setProgress(successAtempts);
        pb.setMax(failedAtempts + successAtempts);
    }

    @Override
    public void onClick(View v)
    {
        if(v == mBtnSuccess)
        {
            GlobalModel.getWpModel().setSuccessAtempts(
                    GlobalModel.getWpModel().getSuccessAtempts() + 1
            );

            setData(GlobalModel.getWpModel().getSuccessAtempts(), GlobalModel.getWpModel().getFailedAtempts());
        }
        if(v == mBtnFailed)
        {
            GlobalModel.getWpModel().setFailedAtempts(
                    GlobalModel.getWpModel().getFailedAtempts() + 1
            );

            setData(GlobalModel.getWpModel().getSuccessAtempts(), GlobalModel.getWpModel().getFailedAtempts());
        }
        if(v == mBtnClear)
        {
            GlobalModel.getWpModel().setFailedAtempts(0);
            GlobalModel.getWpModel().setSuccessAtempts(0);
            setData(0, 0);
        }
    }

    private Button mBtnSuccess;
    private Button mBtnFailed;
    private Button mBtnClear;
}

File: minion/app/src/main/java/snakesoft/minion/Activities/WpActivity.kt
package snakesoft.minion.Activities

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ProgressBar
import android.widget.TextView

import snakesoft.minion.Models.GlobalModel
import snakesoft.minion.R

class WpActivity : AppCompatActivity(), View.OnClickListener {

    private var mBtnSuccess: Button? = null
    private var mBtnFailed: Button? = null
    private var mBtnClear: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_wp)

        val SuccessAttempts = GlobalModel.WpModel.SuccessAttempts
        val FailedAttempts = GlobalModel.WpModel.FailedAttempts

        setData(SuccessAttempts, FailedAttempts)

        mBtnSuccess = findViewById(R.id.btnSuccess) as Button
        mBtnSuccess!!.setOnClickListener(this@WpActivity)

        mBtnFailed = findViewById(R.id.btnFail) as Button
        mBtnFailed!!.setOnClickListener(this@WpActivity)

        mBtnClear = findViewById(R.id.btnClear) as Button
        mBtnClear!!.setOnClickListener(this@WpActivity)
    }

    private fun setData(SuccessAttempts: Int, FailedAttempts: Int) {
        val tv = findViewById(R.id.tvScore) as TextView
        tv.text = Integer.toString(SuccessAttempts) + ":" + Integer.toString(FailedAttempts)

        val pb = findViewById(R.id.progressBar) as ProgressBar
        pb.progress = SuccessAttempts
        pb.max = FailedAttempts + SuccessAttempts
    }

    override fun onClick(v: View) {
        if (v === mBtnSuccess) {
            GlobalModel.WpModel.SuccessAttempts = GlobalModel.WpModel.SuccessAttempts + 1

            setData(GlobalModel.WpModel.SuccessAttempts, GlobalModel.WpModel.FailedAttempts)
        }
        if (v === mBtnFailed) {
            GlobalModel.WpModel.FailedAttempts = GlobalModel.WpModel.FailedAttempts + 1

            setData(GlobalModel.WpModel.SuccessAttempts, GlobalModel.WpModel.FailedAttempts)
        }
        if (v === mBtnClear) {
            GlobalModel.WpModel.FailedAttempts = 0
            GlobalModel.WpModel.SuccessAttempts = 0
            setData(0, 0)
        }
    }
}
